modules/core/src/main/scala/org/tessellation/infrastructure/dag/DAGService.scala

```diff
@@ -55,5 +55,20 @@ object DAGService {
 
       }
 
+      def getTotalWallets: F[Option[(Int, SnapshotOrdinal)]] =
+        globalSnapshotStorage.head.flatMap {
+          _.flatTraverse(snapshot => getTotalWallets(snapshot.value.ordinal))
```

Comment:
You are fetching the same snapshot twice here. Maybe you should just extract the part retrieving needed fields into a separate function and use it in both `getTotalWallets` methods.